pkg/asset/installconfig/gcp/validation.go

```diff
@@ -462,6 +465,26 @@ func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path
 	return allErrs
 }
 
+func validateServiceEndpoints(_ API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// attempt to resolve all the custom (overridden) endpoints. If any are not reachable,
+	// then the installation should fail not skip the endpoint use.
+	for id, serviceEndpoint := range ic.GCP.ServiceEndpoints {
+		// Ignore e2e.local for unit tests.
+		if serviceEndpoint.URL == "e2e.local" {
+			logrus.Warnf("Skipping service endpoint validation for %s", serviceEndpoint.Name)
+			continue
+		}
```

Comment:
I won't block on this, particularly as it is the same approach used for aws, but as a matter of principal it's not a good idea to adjust code specifically for unit tests. Instead, what we should have done for aws is essentially inject a mock http client.